src/modelbench/run.py

```diff
@@ -130,45 +139,61 @@ def find_suts_for_sut_argument(sut_args: List[str]):
     return suts
 
 
-def score_benchmarks(benchmarks, suts, max_instances, debug, parallel=True):
+def score_benchmarks(benchmarks, suts, max_instances, machine_reports=False, parallel=True):
     secrets = load_secrets_from_config()
+
+    # Count total number of tests * SUTs to run.
+    total = 0
+    for b in benchmarks:
+        for h in b.hazards():
+            total += len(h.tests(secrets=secrets))
+    total *= len(suts)
+
     if parallel:
-        f = functools.partial(score_a_sut, benchmarks, max_instances, secrets, debug)
-        with Pool(len(suts)) as p:
-            results = p.map(f, suts)
+        with Manager() as manager:
+            shared_count = manager.Value(ctypes.c_double, 0.0)
+            lock = manager.Lock()
+            progress = ProgressTracker(total, machine_reports, shared_count, lock)
+            f = functools.partial(score_a_sut, benchmarks, max_instances, secrets, progress)
+            with Pool(len(suts)) as p:
+                results = p.map(f, suts)
+                p.close()
+                p.join()
             return list(itertools.chain(*results))
     else:
+        progress = ProgressTracker(total, machine_reports)
         benchmark_scores = []
         for sut in suts:
-            sut_scores = score_a_sut(benchmarks, max_instances, secrets, debug, sut)
+            sut_scores = score_a_sut(benchmarks, max_instances, secrets, progress, sut)
             benchmark_scores.extend(sut_scores)
         return benchmark_scores
 
 
-def score_a_sut(benchmarks, max_instances, secrets, debug, sut):
+def score_a_sut(benchmarks, max_instances, secrets, progress, sut):
     sut_scores = []
-    echo(termcolor.colored(f'Examining system "{sut.display_name}"', "green"))
+    logging.info(termcolor.colored(f'Examining system "{sut.display_name}"', "green"))
     sut_instance = sut.instance(secrets)
     for benchmark_definition in benchmarks:
-        echo(termcolor.colored(f'  Starting run for benchmark "{benchmark_definition.name()}"', "green"))
+        logging.info(termcolor.colored(f'  Starting run for benchmark "{benchmark_definition.name()}"', "green"))
         hazard_scores = []
         for hazard in benchmark_definition.hazards():
-            echo(termcolor.colored(f'    Examining hazard "{hazard.name()}"', "green"))
-
+            logging.info(termcolor.colored(f'    Examining hazard "{hazard.name()}"', "green"))
             results = {}
             for test in hazard.tests(secrets=secrets):
                 items = max_instances
                 results[test.uid] = run_prompt_response_test(
-                    test=test, sut=sut_instance, data_dir="./run", max_test_items=items
+                    test=test,
+                    sut=sut_instance,
+                    data_dir="./run",
```

Comment:
This may be a personal preference, but instead of literals deep down in the code, I recommend using named constants somewhere up top, e.g. 

```python
DEFAULT_DATA_DIR = "./run"

...

run_prompt_response_test(data_dir=DEFAULT_DATA_DIR, ...)
```

This avoids having to replace the literal in more than one place, or having to plow through code to see where exactly that default path is set.